docs/architecture/ADR-002-universal-share-encoding.md

```diff
@@ -0,0 +1,161 @@
+# ADR 009: Universal Share Encoding
+<!-- disable markdownlint MD010 because Go code snippet uses tabs -->
+<!-- markdownlint-disable MD010 -->
+
+## Changelog
+
+- 2022/9/22: inital draft of InfoReservedByte
+- 2022/9/24: update draft to Universal Share Encoding
+
+## Context
+
+- **nid** (8 bytes): namespace id
+- **reserved** (1 byte): is the location of the first transaction, ISR, or evidence in this share if there is one and `0` if there isn't one
+- **message length** (varint 1 to 10 bytes): is the length of the entire message in bytes
+
+The current reserved namespace (transaction, ISRs, evidence) share format is:<br>`nid (8 bytes) | reserved (1 byte) | share data`
+
+The current unreserved namespace (message) share format is:
+
+- First share of message:<br>`nid (8 bytes) | message length (varint 1 to 10 bytes) | share data`
+- Contiguous share in message:<br>`nid (8 bytes) | share data`
+
+The current share format poses multiple challenges:
+
+1. Clients must have two share parsing implementations (one for reserved namespace shares and one for unreserved namespace shares).
```

Comment:
Clarifying question: in this context does "raw message data" apply to the raw data for message shares and for reserved namespace shares (i.e. transactions)?

In other words, is "message" applicable to message shares and for reserved namespace shares? I was confused by the "message" terminology used in https://github.com/celestiaorg/celestia-core/issues/839 because I've been thinking of the two types of shares as different and "message" only applying to message shares. 

> for shares at the start of a message;

Reply:
Yes, by "raw message data" I also mean the bytes in the reserved namespace that fall outside of the "universal share format". Not sure what is a better terminology.

Reply:
Proposal: "raw share data"

Related: https://github.com/celestiaorg/celestia-core/issues/848

Reply:
Hmm share data might also be confusing, because to me that sounds like it's the entire share, because in erasure coding terminology share means the entire share.

Reply:
I see. Given that context, "share data" isn't accurate because it describes all of the share and the term we're looking for only applies to the portion of the share that excludes the universal prefix. 

Alternative proposals: 
- share contents
- share suffix

Reply:
Resolving in favor of https://github.com/celestiaorg/celestia-core/issues/848